src/vm_cpu.rs

```diff
@@ -338,8 +340,83 @@ impl VmPmuState {
             }
         }
     }
-}
 
+    /// Saves the internal state for PMU counters. Stops started counters, and resets all configured counters.
+    /// This should be called in anticipation of an outbound context switch.
+    pub fn save_counters(&mut self) {
+        use PmuCounterState::*;
+        if let Ok(pmu_info) = pmu::PmuInfo::get() {
+            for (i, state) in self.counter_state.iter_mut().enumerate() {
+                if matches!(state, Configured(_) | Started(_)) {
+                    // Unwrap ok: Success is guaranteed
```

Comment:
`unwrap()` comment should say _why_ success is guaranteed (e.g. because the counter must be valid if it was started/configured).